model/service.go

```diff
@@ -54,6 +54,11 @@ type Service struct {
 	// service DNS name.  External services are name-based solution to represent
 	// external service instances as a service inside the cluster.
 	ExternalName string `json:"external"`
+
+	// For the service that runs on VMs, ServiceAccountsOnVm specifies the service
+	// accounts provisioned to these VMs. When multiple service accounts are
+	// specified, use comma as the delimiter.
+	ServiceAccountsOnVm string `json:"serviceaccountsonvm,omitempty"`
```

Comment:
The purpose of this is to serve secure naming, to indicate which service account runs which service on VMs. Because VMs are not like K8s pods that can be tracked by apiserver, secure naming info can't be resolved through the service -> pod and pod -> service account mappings. So we need to explicitly store the service accounts that are used to run the on-VM services. The only matter is how we store it. Currently we've agreed to store it in service annotations.

I admit this doesn't look nice. I think I can change this to just store the annotations, and move the GetServiceAccountsOnVm logic to platform/kube.